assets/js/translator.js

```diff
@@ -581,32 +581,31 @@ function translateText() {
             }
             request.q = $('#originalText').val(); // eslint-disable-line id-length
             request.markUnknown = $('#markUnknown').prop('checked') ? 'yes' : 'no';
-            textTranslateRequest = $.jsonp({
-                url: config.APY_URL + endpoint,
-                beforeSend: ajaxSend,
-                complete: function () {
-                    ajaxComplete();
-                    textTranslateRequest = undefined;
-                },
-                data: request,
-                success: function (data) {
-                    if(data.responseStatus === HTTP_OK_CODE) {
-                        $('#translatedText').val(data.responseData.translatedText);
-                        $('#translatedText').removeClass('notAvailable text-danger');
-                    }
-                    else {
-                        translationNotAvailable();
-                    }
-                },
-                error: translationNotAvailable
-            });
+            var methodType = (request.q.length > THRESHOLD_REQUEST_LENGTH) ? 'POST' : 'GET';
+            var callbackSuccess = handleTranslateSuccessResponse;
+            var callbackError = handleTranslateErrorResponse;
+            ajaxCallForAllModes(methodType, request, endpoint, callbackSuccess, callbackError,textTranslateRequest);
         }
         else {
             translationNotAvailable();
         }
     }
 }
 
+function handleTranslateSuccessResponse(data) {
+    if(data.responseStatus === HTTP_OK_CODE) {
+        $('#translatedText').val(data.responseData.translatedText);
+        $('#translatedText').removeClass('notAvailable text-danger');
+    }
+    else {
+        translationNotAvailable();
+    }
+}
+
+function handleTranslateErrorResponse(xOptions, error) {
+    translationNotAvailable();
```

Comment:
Yep! Makes sense! I wonder what I was thinking while writing this :P . Modified the code :D.